core/drivers/scmi-msg/platform_weaks_a32.S

```diff
@@ -0,0 +1,71 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2019, Linaro Limited
+ */
+#include <arm32_macros.S>
+#include <asm.S>
+#include <drivers/scmi.h>
+#include <kernel/unwind.h>
+
+	/*
+	 * Weak functions for SCMI message driver API the platform does
+	 * not implement.
+	 */
+
+#ifdef CFG_SCMI_MSG_CLOCK
+	.weak plat_scmi_clock_count
+	.weak plat_scmi_clock_get_name
+	.weak plat_scmi_clock_rates_array
+	.weak plat_scmi_clock_rates_by_step
+	.weak plat_scmi_clock_get_current_rate
+	.weak plat_scmi_clock_set_current_rate
+	.weak plat_scmi_clock_get_state
+	.weak plat_scmi_clock_set_state
+#endif
+
+#ifdef CFG_SCMI_MSG_RESET_DOMAIN
+	.weak plat_scmi_rd_count
+	.weak plat_scmi_rd_get_name
+	.weak plat_scmi_rd_autonomous
+	.weak plat_scmi_rd_set_state
+#endif
+
+/* Unique weak function for unsupported handlers returning SCMI_NOT_SUPPORTED */
+FUNC plat_scmi_weak_not_supported , :
+
+#ifdef CFG_SCMI_MSG_CLOCK
+plat_scmi_clock_set_current_rate:
```

Comment:
Wouldn't it be possible to implement all this in C by using some clever `attributes()` instead?

Reply:
not addressed yet...

Reply:
Done. Indeed my idea of aggregating all weak functions into a very few number of function (here only 2) is not that useful. It aimed at optimizing resident memory when pager is used, but that saved only few bytes.
I pushed fixup commits where weak functions are defined in the related scmi protocol driver.